ballerina/metadata_types.bal

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents the metadata associated with a field on an entity.
+# Only used by the generated persist clients and `persist:SQLClient`.
+#
+# + columnName - The name of the SQL table column to which the field is mapped. `()` if the field is not present in the table
+# + 'type - The data-type of the field  
+# + autoGenerated - If true, the value of the field is not defined on creation, 
+#                   but is instead generated by the SQL database  
+# + relation - If the field is from another entity, the relational metadata associated with it
+public type FieldMetadata record {|
+    string? columnName = ();
+    typedesc 'type;
+    boolean autoGenerated = false;
+    RelationMetadata? relation = ();
+|};
+
+# Represents the metadata associated with a relation.
+# Only used by the generated persist clients and `persist:SQLClient`.
+#
+# + entityName - The name of the entity represented in the relation  
+# + refTable - The name of the SQL table which contains the referenced column 
+# + refField - The name of the field in `entityName` that is referenced
+public type RelationMetadata record {|
+    string entityName;
+    string refTable;
+    string refField;
+|};
+
+# Represents the metadata associated with performing an SQL `JOIN` operation.
+# Only used by the generated persist clients and `persist:SQLClient`.
+#
+# + entity - The name of the entity that is being joined  
+# + fieldName - The name of the field in `entity` that is being joined  
```

Comment:
```suggestion
# + fieldName - The name of the field in the `entity` that is being joined  
```

Reply:
Fixed in [83ba8ac](https://github.com/ballerina-platform/module-ballerina-persist/pull/29/commits/83ba8aca943aaaaffd88f25e28fe62a5c00c1ef6)